test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/Listeners/ServiceBusListenerTests.cs

```diff
@@ -33,20 +48,26 @@ public ServiceBusListenerTests()
             MessageReceiver messageReceiver = new MessageReceiver(_testConnection, _entityPath);
             _mockMessageProcessor = new Mock<MessageProcessor>(MockBehavior.Strict, messageReceiver, messageOptions);
 
-            ServiceBusOptions config = new ServiceBusOptions
+            _serviceBusOptions = new ServiceBusOptions
             {
                 MessageHandlerOptions = messageOptions
             };
-            _mockMessagingProvider = new Mock<MessagingProvider>(MockBehavior.Strict, new OptionsWrapper<ServiceBusOptions>(config));
+            _mockMessagingProvider = new Mock<MessagingProvider>(MockBehavior.Strict, new OptionsWrapper<ServiceBusOptions>(_serviceBusOptions));
 
             _mockMessagingProvider.Setup(p => p.CreateMessageProcessor(_entityPath, _testConnection))
                 .Returns(_mockMessageProcessor.Object);
 
-            ServiceBusTriggerExecutor triggerExecutor = new ServiceBusTriggerExecutor(_mockExecutor.Object);
-            var mockServiceBusAccount = new Mock<ServiceBusAccount>(MockBehavior.Strict);
-            mockServiceBusAccount.Setup(a => a.ConnectionString).Returns(_testConnection);
+            _triggerExecutor = new ServiceBusTriggerExecutor(_mockExecutor.Object);
+            _mockServiceBusAccount = new Mock<ServiceBusAccount>(MockBehavior.Strict);
+            _mockServiceBusAccount.Setup(a => a.ConnectionString).Returns(_testConnection);
+
+            _loggerFactory = new LoggerFactory();
+            _loggerProvider = new TestLoggerProvider();
+            _loggerFactory.AddProvider(_loggerProvider);
 
-            _listener = new ServiceBusListener(_entityPath, false, triggerExecutor, config, mockServiceBusAccount.Object, _mockMessagingProvider.Object);
+            _listener = new ServiceBusListener(_functionId, EntityType.Queue, _entityPath, false, _triggerExecutor, _serviceBusOptions, _mockServiceBusAccount.Object,
+                                _mockMessagingProvider.Object, _loggerFactory);
+            _scaleMonitor = (ServiceBusScaleMonitor)_listener.GetMonitor();
```

Comment:
Let's move all these new tests into a new ServiceBusScaleMonitorTests class, now that you've factored the logic cleanly out of the listener. In this file, you could just have one new test for the new GetMonitor method.